### DISCUSSION TITLE
Can you get the separator when using the Separated parser?

### DISCUSSION THREAD

**Discussion by deanebarker** [sebastienros/parlot#79] [General]
If I do this...

```
var results = Separated(delimeterParser, itemParser);
```

...how can I get ahold of the `delimterParser`? It can match on one of three things, and I need to use that thing in the `itemParser`. Does it always skip this and throw it away?

I don't _think_ this is possible, so I've actually built the delimeter into the item's parser, but I thought I'd check.

**Comment by sebastienros** [comment]
You can reconstruct this parser by combining explicit ones, like THING.And(ZeorOrMany(SEP.And(THING))). Which implies more complexity to build the AST, but not much more than if the separator was available in the `Separated` parser.

**Comment by deanebarker** [comment]
I was thinking about writing a parser for `CapturedSeparated`, but, logically, how do you determine which "side" the separator "belongs" to? You could specify, I guess, but there will always be an item with no separator -- be it on the beginning or the end.

Not sure how generically applicable this is.
